pkg/apis/duck/v1/resource_types_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2020 The Knative Authors
```

Comment:
Thanks for adding this @danyinggu !!!
I'm realizing that I think are not using resource anymore, and instead are using KResource from pkg.
Can you check if our v1beta1 Resource is being used. If not, then I think we should remove them from both v1alpha1, v1beta1, and do not promote it to v1...
You will have to run update-codegen again after removed

Reply:
Just checked the source code and synced offline with Nacho, the resource is actually used by keda as it is not a KResource, so we need to keep the resource as it.